cmd/aws-vpc-cni/main.go

```diff
@@ -0,0 +1,409 @@
+	envRandomizeSNAT         = "AWS_VPC_K8S_CNI_RANDOMIZESNAT"
+)
+
+func getEnv(env, defaultVal string) string {
+	if val, ok := os.LookupEnv(env); ok {
+		return val
+	}
+	return defaultVal
+}
+
+// NetConfList describes an ordered list of networks.
+type NetConfList struct {
+	CNIVersion string `json:"cniVersion,omitempty"`
+
+	Name         string     `json:"name,omitempty"`
+	DisableCheck bool       `json:"disableCheck,omitempty"`
+	Plugins      []*NetConf `json:"plugins,omitempty"`
+}
+
+// NetConf stores the common network config for the CNI plugin
+type NetConf struct {
+	CNIVersion string `json:"cniVersion,omitempty"`
+
+	Name         string          `json:"name,omitempty"`
+	Type         string          `json:"type,omitempty"`
+	Capabilities map[string]bool `json:"capabilities,omitempty"`
+	IPAM         *IPAMConfig     `json:"ipam,omitempty"`
+	DNS          *types.DNS      `json:"dns,omitempty"`
+
+	RawPrevResult map[string]interface{} `json:"prevResult,omitempty"`
+	PrevResult    types.Result           `json:"-"`
+
+	// Interface inside container to create
+	IfName string `json:"ifName,omitempty"`
+
+	Enabled string `json:"enabled,,omitempty"`
+
+	// IP to use as SNAT target
+	NodeIP net.IP `json:"nodeIP,omitempty"`
+
+	VethPrefix string `json:"vethPrefix,omitempty"`
+
+	PodSGEnforcingMode string `json:"podSGEnforcingMode,omitempty"`
+
+	RandomizeSNAT string `json:"randomizeSNAT,omitempty"`
+
+	// MTU for eth0
+	MTU string `json:"mtu,omitempty"`
+
+	PluginLogFile string `json:"pluginLogFile,omitempty"`
+
+	PluginLogLevel string `json:"pluginLogLevel,omitempty"`
+}
+
+// IPAMConfig references containernetworking structure defined at https://github.com/containernetworking/plugins/blob/main/plugins/ipam/host-local/backend/allocator/config.go
+type IPAMConfig struct {
+	*Range
+	Name       string         `json:"name,omitempty"`
+	Type       string         `json:"type,omitempty"`
+	Routes     []*types.Route `json:"routes,omitempty"`
+	DataDir    string         `json:"dataDir,omitempty"`
+	ResolvConf string         `json:"resolvConf,omitempty"`
+	Ranges     []RangeSet     `json:"ranges"`
+	IPArgs     []net.IP       `json:"-"` // Requested IPs from CNI_ARGS and args
+}
+
+// RangeSet references containernetworking structure
+type RangeSet []Range
+
+// Range references containernetworking structure
+type Range struct {
+	RangeStart net.IP      `json:"rangeStart,omitempty"` // The first ip, inclusive
+	RangeEnd   net.IP      `json:"rangeEnd,omitempty"`   // The last ip, inclusive
+	Subnet     types.IPNet `json:"subnet"`
+	Gateway    net.IP      `json:"gateway,omitempty"`
+}
+
+func waitForIPAM() bool {
+	for {
+		cmd := exec.Command("./grpc-health-probe", "-addr", "127.0.0.1:50051", ">", "/dev/null", "2>&1")
+		var outb bytes.Buffer
+		cmd.Stdout = &outb
+		cmd.Run()
+		if outb.String() == "" {
+			return true
+		}
+	}
+}
+
+// Wait for vpcCniInitDonePath to exist (maximum wait time is 60 seconds)
+func waitForInit() error {
```

Comment:
This function is anyways commented, we can delete it.

Reply:
same as above, this is temporarily commented to simplify rebase for #2137